packages/arb-token-bridge-ui/src/token-bridge-sdk/Erc20TeleportStarter.ts

```diff
@@ -0,0 +1,195 @@
+  }
+
+  public async approveNativeCurrency() {
+    // no-op, but will need to be implemented for custom-native-token teleports later
+  }
+
+  public async approveNativeCurrencyEstimateGas() {
+    // no-op, but will need to be implemented for custom-native-token teleports later
+  }
+
+  public requiresTokenApproval = async ({
+    amount,
+    signer
+  }: RequiresTokenApprovalProps) => {
+    if (!this.sourceChainErc20Address) {
+      throw Error('Erc20 token address not found')
+    }
+
+    const address = await getAddressFromSigner(signer)
+
+    const l1l3Bridger = await this.getBridger()
+
+    const l1TeleporterAddress = l1l3Bridger.teleporterAddresses.l1Teleporter
+
+    const allowanceForTeleporter = await fetchErc20Allowance({
+      address: this.sourceChainErc20Address,
+      provider: this.sourceChainProvider,
+      owner: address,
+      spender: l1TeleporterAddress
+    })
+
+    return allowanceForTeleporter.lt(amount)
+  }
+
+  public async approveTokenEstimateGas({ signer, amount }: ApproveTokenProps) {
+    if (!this.sourceChainErc20Address) {
+      throw Error('Erc20 token address not found')
+    }
+    const address = await getAddressFromSigner(signer)
+
+    const l1l3Bridger = await this.getBridger()
+    const l1TeleporterAddress = l1l3Bridger.teleporterAddresses.l1Teleporter
+
+    const contract = ERC20__factory.connect(
+      this.sourceChainErc20Address,
+      this.sourceChainProvider
+    )
+    return contract.estimateGas.approve(
+      l1TeleporterAddress,
+      amount ?? constants.MaxUint256,
+      {
+        from: address
+      }
+    )
+  }
+
+  public async approveToken({ signer, amount }: ApproveTokenProps) {
+    if (!this.sourceChainErc20Address) {
+      throw Error('Erc20 token address not found')
+    }
+
+    const l1l3Bridger = await this.getBridger()
+
+    return l1l3Bridger.approveToken({
+      erc20L1Address: this.sourceChainErc20Address,
+      l1Signer: signer,
+      amount
+    })
+  }
+
+  public async transferEstimateGas({ amount, signer }: TransferEstimateGas) {
+    if (!this.sourceChainErc20Address) {
+      throw Error('Erc20 token address not found')
+    }
+
+    // get the intermediate L2 chain provider
+    const { l2Provider } = await getL2ConfigForTeleport({
+      destinationChainProvider: this.destinationChainProvider
+    })
+
+    const l1l3Bridger = await this.getBridger()
+
+    const depositRequest = await l1l3Bridger.getDepositRequest({
+      l1Signer: signer,
+      erc20L1Address: this.sourceChainErc20Address,
+      amount,
+      l1Provider: this.sourceChainProvider,
+      l2Provider,
+      l3Provider: this.destinationChainProvider
+    })
+
+    try {
+      const estimatedParentChainGas =
+        await this.sourceChainProvider.estimateGas(depositRequest.txRequest)
+      return {
+        estimatedParentChainGas: percentIncrease(
+          estimatedParentChainGas,
+          BigNumber.from(5)
+        ),
```

Comment:
i see this in the `EthTeleportStarter` too, perhaps we'd want to move this to `BridgeTransferStarter` at some point?

Reply:
yea just suggesting if that's sth we want to do in the future 👍